recipes/android-ndk/all/conanfile.py

```diff
@@ -294,12 +298,25 @@ def package_info(self):
         self.conf_info.define("tools.android:ndk_path", os.path.join(self.package_folder, "bin"))
 
         compiler_executables = {
-            "c": self._define_tool_var("CC", "clang"),
-            "cpp": self._define_tool_var("CXX", "clang++"),
+            "c": self._define_tool_var_naked("CC", "clang"),
+            "cpp": self._define_tool_var_naked("CXX", "clang++"),
         }
+
+        # https://github.com/android/ndk/issues/1856
+        compiler_flags = [f"--target={self._clang_target}"]
```

Comment:
Having a closer look at the issue it is not clear this solves the issue. As it is still in discussion, I would like to know if you have checked this isolated fix by yourself.

Reply:
The issue I encountered closely resembled one I experienced when building my project with the `libsodium` package as a dependency. I traced the problem to the wrapper scripts, as described [here](https://github.com/android/ndk/issues/1856#issuecomment-1542248775), which resolved the issue for me.

I then attempted to modify the `android-ndk` recipe so I wouldn't have to manually adjust the NDK scripts on my machine. I tested the build both without my changes in the recipe (where it failed) and with them (where it succeeded). I've been using the `android-ndk` package with these modifications for several months now and haven't faced any issues. However, I should note that I only use it for cross-compiling from Windows to Android.

If necessary, I can provide a minimal reproducible example to demonstrate the issue.